src/zabapgit_objects.prog.abap

```diff
@@ -155,6 +165,161 @@ CLASS lcl_objects_activation IMPLEMENTATION.
 
   ENDMETHOD.                    "activate
 
+
+  METHOD use_new_activation_logic.
+
+    IF zcl_abapgit_persist_settings=>get_instance( )->read( )->get_experimental_features( ) = abap_true.
+
+      CALL FUNCTION 'FUNCTION_EXISTS'
+        EXPORTING
+          funcname           = 'DD_MASS_ACT_C3'    " Name of Function Module
+        EXCEPTIONS
+          function_not_exist = 1
+          OTHERS             = 2.
+
+      IF sy-subrc = 0.
+        rv_use_new_activation_logic = abap_true.
+      ENDIF.
+
+    ENDIF.
+
+  ENDMETHOD.
+
+
+  METHOD activate_new.
+
+    IF gt_objects IS INITIAL.
+      RETURN.
+    ENDIF.
+
+    IF iv_ddic = abap_true.
+
+      zcl_abapgit_progress=>show( iv_key     = 'Activating DDIC'
+                                  iv_current = '98'
+                                  iv_total   = '100'
+                                  iv_text    = '...' ).
+
+      activate_ddic( ).
+
+    ELSE.
+
+      zcl_abapgit_progress=>show( iv_key     = 'Activating non DDIC'
+                                  iv_current = '98'
+                                  iv_total   = '100'
+                                  iv_text    = '...' ).
+
+      activate_non_ddic( ).
```

Comment:
perhaps just call `activate_old( abap_false )`, then method `activate_non_ddic` can be deleted

Reply:
good point, changed...